tests/conftest.py

```diff
@@ -1852,6 +1853,96 @@ def bucket_cleanup():
     return _create_buckets
 
 
+@pytest.fixture()
+def bucket_class_factory(request, mcg_obj, backingstore_factory):
+    """
+    Create a bucket class factory. Calling this fixture creates a new custom bucket class.
+    For a custom backingstore(s), provide the 'backingstore_dict' parameter.
+
+    Args:
+        mcg_obj_session (MCG): An MCG object containing the MCG S3 connection credentials
+
+    """
+    interfaces = {
+        'oc': mcg_obj.oc_create_bucketclass,
+        'cli': mcg_obj.cli_create_bucketclass
+    }
+    created_bucket_classes = []
+
+    def _create_bucket_class(
+        bucket_class_dict
+    ):
+        """
+        Creates and deletes all bucket classes that were created as part of the test
+
+        Args:
+
+            bucket_class_dict (dict): Dictionary containing the description of the bucket class.
+                possible keys and values are:
+                - interface (str): The interface to use for creation of buckets.
+                    OC | CLI
+                - backingstore_dict (dict/list): Either existing list of backingstores or
+                    A dictionary compatible with the backing store factory requirements.
+                - placement (str): The Placement policy for this bucket class.
+                    Spread | Mirror
+                if no key is provided default values will apply.
+
+        Returns:
+            list: A Bucket Class object.
+
+        """
+        if 'interface' in bucket_class_dict:
+            interface = bucket_class_dict['interface']
+            if interface.lower() not in interfaces.keys():
+                raise RuntimeError(
+                    f'Invalid interface type received: {interface}. '
+                    f'available types: {", ".join(interfaces)}'
+                )
+            else:
+                interface = bucket_class_dict['interface']
+        else:
+            interface = 'OC'
+        if 'backingstore_dict' in bucket_class_dict:
+            if isinstance(bucket_class_dict['backingstore_dict'], dict):
+                backingstores = [backingstore.name for backingstore in backingstore_factory(
+                    interface, bucket_class_dict['backingstore_dict'])]
+            else:
+                backingstores = [backingstore.name for backingstore in bucket_class_dict['backingstores']]
```

Comment:
not sure it's optimal to accept 2 types for the same arg. I would suggest adjusting the usage of this function and send a single type (either dict or some other class)

Reply:
i wanted to allow additional versatility allowing the usage of existing backing store but it can be removed as well

Reply:
I think it would be more robust to accept one type

Reply:
ok so i'll change it to accept only a dictionary and if no dictionary is given then the default backing store will be used